test/topology_custom/test_mv_topology_change.py

```diff
@@ -193,3 +194,56 @@ async def test_mv_write_to_dead_node(manager: ManagerClient):
     # will be held for long time until the write timeouts.
     # Otherwise, it is expected to complete in short time.
     await manager.remove_node(servers[0].server_id, servers[-1].server_id, timeout=30)
+
+# Reproduces https://github.com/scylladb/scylladb/issues/21492
+# In this test we perform a write with a view update while the replication factor
+# of the keyspace is being changed and the corresponding base tablet finishes
+# the migration at a different time the view tablet does.
+@pytest.mark.asyncio
+@pytest.mark.parametrize("delayed_replica", ["base", "view"])
+@pytest.mark.parametrize("altered_dc", ["dc1", "dc2"])
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_mv_rf_change(manager: ManagerClient, delayed_replica: str, altered_dc: str):
+    # The workaround for this issue depends on the fact that only last replica
+    # in the DCs replica list is modified when changing RF. Use 2 DCs to ensure
+    # that the fix works even if the new replica is not the last one in the total list.
+    servers = await manager.servers_add(2, property_file={'dc': f'dc1', 'rack': 'myrack'})
+    servers.extend(await manager.servers_add(2, property_file={'dc': f'dc2', 'rack': 'myrack'}))
+
+    cql = manager.get_cql()
+    await cql.run_async("CREATE KEYSPACE IF NOT EXISTS ks WITH replication = {'class': 'NetworkTopologyStrategy', 'dc1': 1, 'dc2': 1} AND tablets = {'initial': 1}")
```

Comment:
What if the `ks` exists and have different topology, won't that break the test? Perhaps it's better to generate a random name for the keyspace.

Reply:
I don't think we reuse clusters in the `topology` suite, so this shouldn't be a problem